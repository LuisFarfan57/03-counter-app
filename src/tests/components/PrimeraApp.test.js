import PrimeraApp from "../../components/PrimeraApp"
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'

describe('Pruebas en PrimeraApp', () => {
    // test('', () => {
    //     const saludo = 'Luis'

    //     const wrapper = render(<PrimeraApp  saludo={saludo} />)

    //     expect(wrapper.getByText(saludo)).toBeInTheDocument()
    // })

    test('debe de mostrar componente PrimeraApp correctamente', () => {
        const saludo = 'Luis'

        const wrapper = shallow(<PrimeraApp saludo={saludo} />)

        expect(wrapper).toMatchSnapshot()
    })

    test('debe de mostrar el subtitulo correctamente', () => {
        const saludo = 'Luis'
        const subtitulo = 'Soy un subtitulo desde test'

        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />)

        //Parecido al querySelector
        const textoParrafo = wrapper.find('p').text()

        expect(textoParrafo).toBe(subtitulo)
    })
})