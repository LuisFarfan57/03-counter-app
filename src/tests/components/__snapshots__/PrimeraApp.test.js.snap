// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe de mostrar componente PrimeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Luis
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
